Swabbing improvements, Swab Applicators and Synthetic Swabs

Mehnix opened the pull request:
Adds new items to aid in the Botanist's crossbreeding endeavours, and makes it a bit faster.

## About the PR
Adds Swab Applicators, a dirty swab slotted into one of these will not be cross contaminated with future plants it is swabbed on, and so can apply its original swabbed pollon infinitely.
Adds Synthetic Swabs, a swab that can be cleaned by interacting with it, allowing for reuse. It is also easier to make than regular swabs.
Both of these items are researched under Tier 1 Hydroponics by science.

Swabbing has been made faster, taking 1 second rather than 2 because i've lost enough of my life to the 2 second doafter bar. It also has a sound now.

## Why / Balance
There's not much that science can research that directly benefits botany. This adds two new things that are useful but not essential to the botany experience.

Swab applicators allow keeping the spirit of a plant alive even after it dies, with a single swab able to be infinitely applied onto future plants at will.

Meanwhile you'll never run out of swabs with synthetics able to be cleaned and reused with ease.

## Technical details
Two new types of swab, modifications to the existing Botany Swab system.

## Media
https://github.com/user-attachments/assets/302d780b-8960-4c47-b978-0b08e523b050

## Requirements
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
None, Probably

**Changelog**
:cl:
- add: Added Researchable Synthetic Swabs, reusable swabs.
- add: Added Researchable Swab Applicators, allowing repeated applying of a swab without risk of cross-contamination.
- tweak: Swabbing is faster.

Mehnix committed:
We Swabbin

iaada commented on Content.Server/Botany/Components/BotanySwabComponent.cs:
```diff
@@ -9,7 +10,38 @@ namespace Content.Server.Botany
+
+        /// <summary>
+        /// Whether the swab can be used if it has no seed data, default true
+        /// If false, a seperate may to provide seed data is required or the swab will be unusable
```
```suggestion
        /// If false, a seperate way to provide seed data is required or the swab will be unusable
```

Mehnix replied:
✓

iaada commented on Content.Server/Botany/Systems/BotanySwabSystem.cs:
```diff
@@ -71,12 +88,55 @@ private void OnDoAfter(EntityUid uid, BotanySwabComponent swab, DoAfterEvent arg
+    private void OnInsertAttempt(EntityUid uid, BotanySwabComponent swab, ref ContainerGettingInsertedAttemptEvent args)
+    {
+        //does the container have the botanySwab component
+        if (!HasComp<BotanySwabComponent>(args.Container.Owner))
```
Wouldn't this make it impossible to put swabs into a container? I think you need to change your logic here.
```suggestion
        if (swab.SeedData == null && HasComp<BotanySwabComponent>(args.Container.Owner))
```

Mehnix replied:
The function cancels if its criteria of the owner (the swab applicator) not having the BotanySwabComponent and the swab not being null aren't met. It returning means that the insert attempt "passes" and continues as it should.

In game this is seen with trying to put a clean swab into a swab applicator causing it to pop back out due to clean swabs not being accepted. Used ones will be accepted just fine.

iaada commented on Content.Server/Botany/Components/BotanySwabComponent.cs:
```diff
@@ -9,7 +10,38 @@ namespace Content.Server.Botany
+        /// If false, a seperate may to provide seed data is required or the swab will be unusable
+        /// </summary>
+        [DataField]
+        public bool Usable = true;
```
I feel this variable needs a more descriptive name.

Mehnix replied:
changed to UseIfClean

Mehnix committed:
redundant line

zeroVisible doesn't exist any more

Mehnix committed:
Updates based on comments

Mehnix committed:
e

Mehnix committed:
ejectOnUse

fix yaml fail by performing the advanced task of properly reading the ItemSlotsComponent layout

Cojoke-dot commented:
Off topic(I like the idea of the swabs) but is there something in the guidebook explaining how reusing the regular swabs work? It could get confusing with more types of swabs thrown into a shift when someone tries it out for the first time.

Mehnix commented:
> Off topic(I like the idea of the swabs) but is there something in the guidebook explaining how reusing the regular swabs work? It could get confusing with more types of swabs thrown into a shift when someone tries it out for the first time.

Not currently, if this PR gets merged i'd be fine with writing that myself as swab crossbreeding and cross-contamination isn't particularly intuitive. Some time last year I ended up just to reading the code to figure out its specifics.

Mehnix committed:
Merge branch 'master' into MoreSwab

Mehnix committed:
Merge branch 'master' into MoreSwab

Mehnix committed:
Let swabs exist in belts

and also swab applicator fits better in 1x2

Mehnix committed:
belt but works hopefully

Mehnix commented on Content.Server/Botany/Systems/BotanySwabSystem.cs:
```diff
@@ -12,13 +16,18 @@ public sealed class BotanySwabSystem : EntitySystem
         SubscribeLocalEvent<BotanySwabComponent, ExaminedEvent>(OnExamined);
         SubscribeLocalEvent<BotanySwabComponent, AfterInteractEvent>(OnAfterInteract);
         SubscribeLocalEvent<BotanySwabComponent, BotanySwabDoAfterEvent>(OnDoAfter);
+        SubscribeLocalEvent<BotanySwabComponent, UseInHandEvent>(OnClean); //Remove a swab's SeedData
```
Moved the comments to function summaries

Mehnix committed:
Updates based on comments

Mehnix commented on Content.Server/Botany/Systems/BotanySwabSystem.cs:
```diff
@@ -44,6 +53,12 @@ private void OnAfterInteract(EntityUid uid, BotanySwabComponent swab, AfterInter
         if (args.Target == null || !args.CanReach || !HasComp<PlantHolderComponent>(args.Target))
             return;
 
+        if (swab.UsableIfClean == false && swab.SeedData == null)
```
personally I think == true/false is better for code readability but it's not a hill i die on, so, sure.

Mehnix commented on Content.Server/Botany/Systems/BotanySwabSystem.cs:
```diff
@@ -31,7 +40,7 @@ private void OnExamined(EntityUid uid, BotanySwabComponent swab, ExaminedEvent a
             if (swab.SeedData != null)
                 args.PushMarkup(Loc.GetString("swab-used"));
-            else
+            else if (swab.UsableIfClean == true)
```
same as above

Mehnix committed:
Merge branch 'master' into MoreSwab

Mehnix committed:
Merge branch 'master' into MoreSwab
